README.md

```diff
@@ -52,6 +51,74 @@ To reload the extension after some changes, open [`chrome://extensions`](chrome:
+
+If you're new to developing browser extensions, there are a couple of crucial parts you need to be aware of:
+
+1. **Manifest File** - Located at `src/manifest.json`. This file instructs the browser when to run which scripts (for example, "on a page that matches a GitHub pull request, run the github entrypoint"). If you plan to extend our extension's capabilities to new sites, you'll have to update this file accordingly.
+
+2. **Background Runner** - The `src/background.js` file handles requests to the Orbit API in a separate thread, ensuring smooth performance. Look for instances of `chrome.runtime.sendMessage` to see examples of how this communication is performed.
+
+## The Code Flow
+
+There exists a dedicated entrypoint for each supported site, like `src/widget/githubEntrypoint.js`. This script is responsible for determining the right moment to attempt loading the widget.
+
+During initialization, the entrypoint script also sets up an array of `Page` objects. Each `Page` represents a specific type of page on the website where our widget can load. They handle all the site-specific logic. For instance, to confirm we're on a particular type of page, a `Page` object would use the `#detect()` method. Or, to figure out where to place our widget, it would use `#findInsertionPoint()`. You can get more insights into these functions in the superclass at `src/pages/page.js`.
+
+The entry point's final task is to kickstart the `src/widget/widgetOrchestrator.js`. This script checks if our widget can load on the current page and inserts a button, the widget, and a slot for additional data in appropriate zones on the supported sites.
+
+Lastly, most of the remaining logic takes place in `src/components/widget.js`, which handles interactions with the Orbit API and manages the widget's various display states.
+
+## Adding Support for a New Site
+
+**For an example of this process being followed: https://github.com/orbit-love/orbit-browser-extension/pull/45**
+
+To add support for a new site to the extension:
+
+1. Create a New Entrypoint
+
+Start by creating a new entry point file, similar to src/widget/githubEntrypoint.js. This will require understanding when to load the widget for this specific site. Broadly speaking, `DOMContentLoaded` is a good place to start. However different sites have different requirements, so it will require some investigation to identify the appropriate events to observe for the page load lifecycle.
+
+In creating this, you'll reach a point where you need to define the pages to pass to the widget orchestrator. So...
+
+2. Create a New Page Class
+
+Create a new Page class for the new site by extending `src/pages/page.js`. This class will handles page-specific logic - see the comments in the superclass for details on the functions to implement, and take inspiration from the existing types of page.
```

Comment:
is it “types of page” or “types of pages”?

Reply:
I think the latter 😅 That seems more correct at least, thnaks!